xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-notifications/src/test/java/org/xwiki/mentions/templates/MentionTest.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.mentions.templates;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.velocity.VelocityContext;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.xwiki.eventstream.internal.DefaultEvent;
+import org.xwiki.mentions.internal.MentionView;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.notifications.CompositeEvent;
+import org.xwiki.platform.date.script.DateScriptService;
+import org.xwiki.script.service.ScriptService;
+import org.xwiki.template.TemplateManager;
+import org.xwiki.test.page.HTML50ComponentList;
+import org.xwiki.test.page.IconSetup;
+import org.xwiki.test.page.PageTest;
+import org.xwiki.test.page.ScriptServiceComponentList;
+import org.xwiki.test.page.XHTML10ComponentList;
+import org.xwiki.velocity.VelocityManager;
+import org.xwiki.velocity.tools.EscapeTool;
+
+import com.xpn.xwiki.doc.XWikiDocument;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalToCompressingWhiteSpace;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests of the mention notifications template ({@code mention.vm}).
+ * <p>
+ * This test suite assert the html rendered by the mention notification template.
+ *
+ * @version $Id$
+ * @since 13.4RC1
+ */
+@HTML50ComponentList
+@XHTML10ComponentList
+@ScriptServiceComponentList
+class MentionTest extends PageTest
+{
+    /**
+     * Needs to be mocked because don't want the test result to be dependent of the current time.
+     */
+    @Mock
+    private DateScriptService dateScriptService;
+
+    @BeforeEach
+    void setUp() throws Exception
+    {
+        // Initializes then environment for the icon extension.
+        IconSetup.setUp(this.xwiki, this.context, this.oldcore);
```

Comment:
Icon environment setup. It could be moved to `PageTest`, to initialize the icons even if they are not always used in the test, to prevent testers to forget to call this initializer.